subitem/client.go

```diff
@@ -0,0 +1,192 @@
+	var body *stripe.RequestValues
+	var commonParams *stripe.Params
+
+	if params != nil {
+		body = &stripe.RequestValues{}
+		params.AppendTo(body)
+		commonParams = &params.Params
+	}
+
+	item := &stripe.SubItem{}
+	err := c.B.Call("GET", fmt.Sprintf("/subscriptions/%v", id), c.Key, body, commonParams, item)
+
+	return item, err
+}
+
+// Update updates a subscription's properties.
+// For more details see https://stripe.com/docs/api#update_subscription.
+func Update(id string, params *stripe.SubItemParams) (*stripe.SubItem, error) {
+	return getC().Update(id, params)
+}
+
+func (c Client) Update(id string, params *stripe.SubItemParams) (*stripe.SubItem, error) {
+	var body *stripe.RequestValues
+	var commonParams *stripe.Params
+	token := c.Key
+
+	if params != nil {
+		body = &stripe.RequestValues{}
+
+		if len(params.Plan) > 0 {
+			body.Add("plan", params.Plan)
+		}
+
+		if params.NoProrate {
+			body.Add("prorate", strconv.FormatBool(false))
+		}
+
+		if params.Quantity > 0 {
+			body.Add("quantity", strconv.FormatUint(params.Quantity, 10))
+		} else if params.QuantityZero {
+			body.Add("quantity", "0")
+		}
+
+		if params.ProrationDate > 0 {
+			body.Add("proration_date", strconv.FormatInt(params.ProrationDate, 10))
+		}
+
+		commonParams = &params.Params
+		params.AppendTo(body)
+	}
+
+	sub := &stripe.Sub{}
+	err := c.B.Call("POST", fmt.Sprintf("/subscriptions_items/%v", id), token, body, commonParams, sub)
+
+	return sub, err
+}
+
+// Cancel removes a subscription.
+// For more details see https://stripe.com/docs/api#cancel_subscription.
+func Del(id string, params *stripe.SubItemParams) (*stripe.SubItem, error) {
+	return getC().Cancel(id, params)
+}
+
+func (c Client) Del(id string, params *stripe.SubItemParams) (*stripe.SubItem, error) {
+	var body *stripe.RequestValues
+	var commonParams *stripe.Params
+
+	if params != nil {
+		body = &stripe.RequestValues{}
+		params.AppendTo(body)
+		commonParams = &params.Params
+	}
+
+	item := &stripe.Sub{}
+	err := c.B.Call("DELETE", fmt.Sprintf("/subscriptions_item/%v", id), c.Key, body, commonParams, item)
+
+	return sub, err
+}
+
+// List returns a list of subscriptions.
+// For more details see https://stripe.com/docs/api#list_subscriptions.
+func List(params *stripe.SubListParams) *Iter {
+	return getC().List(params)
+}
+
+func (c Client) List(params *stripe.SubListParams) *Iter {
+	var body *stripe.RequestValues
+	var lp *stripe.ListParams
+	var p *stripe.Params
+
+	if params != nil {
+		body = &stripe.RequestValues{}
+
+		if len(params.Customer) > 0 {
+			body.Add("customer", params.Customer)
+		}
+
+		if len(params.Plan) > 0 {
+			body.Add("plan", params.Plan)
+		}
+
+		params.AppendTo(body)
+
+		lp = &params.ListParams
+		p = params.ToParams()
+	}
+
+	return &Iter{stripe.GetIter(lp, body, func(b *stripe.RequestValues) ([]interface{}, stripe.ListMeta, error) {
+		list := &stripe.SubList{}
+		err := c.B.Call("GET", "/subscriptions", c.Key, b, p, list)
```

Comment:
@kyleconroy I'm not sure whether this is finished, but this doesn't seems quite right. I believe the new endpoint is at `/subscription_items`, and if so, it means that not only is this wrong, but your test suite is either broken or not exhaustive enough to be able to test for the failure.